### PR TITLE
Fuse Airspeed

### PR DESCRIPTION
**Pull request by CarlOlsson**
Initial implementation of airspeed measurement fusion

### CODE REVIEW

File: EKF/airspeed_fusion.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+ * @author Carl Olsson <carlolsson.co@gmail.com>
+ * @author Roman Bast <bapstroman@gmail.com>
+ * @author Paul Riseborough <p_riseborough@live.com.au>
+ *
+ */
+#include "ekf.h"
+
+void Ekf::fuseAirspeed()
+{
+	void Ekf::fuseAirspeed() {
```
Comment (tumbili) [inline EKF/airspeed_fusion.cpp]: @CarlOlsson Remove this.

File: EKF/airspeed_fusion.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+			Kfusion[17] = SK_TAS[0]*(P[17][3]*SH_TAS[2] - P[17][22]*SH_TAS[2] + P[17][4]*SK_TAS[1] - P[17][23]*SK_TAS[1] + P[17][5]*vd*SH_TAS[0]);
+			Kfusion[18] = SK_TAS[0]*(P[18][3]*SH_TAS[2] - P[18][22]*SH_TAS[2] + P[18][4]*SK_TAS[1] - P[18][23]*SK_TAS[1] + P[18][5]*vd*SH_TAS[0]);
+			Kfusion[19] = SK_TAS[0]*(P[19][3]*SH_TAS[2] - P[19][22]*SH_TAS[2] + P[19][4]*SK_TAS[1] - P[19][23]*SK_TAS[1] + P[19][5]*vd*SH_TAS[0]);
+			Kfusion[20] = SK_TAS[0]*(P[20][3]*SH_TAS[2] - P[20][22]*SH_TAS[2] + P[20][4]*SK_TAS[1] - P[20][23]*SK_TAS[1] + P[20][5]*vd*SH_TAS[0]);
+			Kfusion[21] = SK_TAS[0]*(P[21][3]*SH_TAS[2] - P[21][22]*SH_TAS[2] + P[21][4]*SK_TAS[1] - P[21][23]*SK_TAS[1] + P[21][5]*vd*SH_TAS[0]);
+			Kfusion[22] = SK_TAS[0]*(P[22][3]*SH_TAS[2] - P[22][22]*SH_TAS[2] + P[22][4]*SK_TAS[1] - P[22][23]*SK_TAS[1] + P[22][5]*vd*SH_TAS[0]);
+			Kfusion[23] = SK_TAS[0]*(P[23][3]*SH_TAS[2] - P[23][22]*SH_TAS[2] + P[23][4]*SK_TAS[1] - P[23][23]*SK_TAS[1] + P[23][5]*vd*SH_TAS[0]);
+
+			// calculate measurement innovation
+			_airspeed_innov = VtasPred - _airspeed_sample_delayed.airspeed; // This is TAS, maybe we should indicate that in some way
```
Comment (tumbili) [inline EKF/airspeed_fusion.cpp]: @CarlOlsson Let's try to keep code style consistent, e.g. v_tas_pred. I also forget from time to time.

**Comment by priseborough** [comment]
Before merging this we need:

1) Protection for a badly conditioned innovation variance calculation and div0
2) Innovation consistency check with adjustable gate parameter
3) logic to set Kalman gains to zero for inactive mag field states
4) flight test data showing airspeed innovations and wind velocity estimates

**Comment by tumbili** [comment]
@CarlOlsson Carl, could you please rebase this on master?

**Comment by priseborough** [comment]
replaced with #75

**Pull request closed without merging**